blockchain/chain/src/store/chain_store.rs

```diff
@@ -93,12 +94,17 @@ impl<DB> ChainStore<DB>
 where
     DB: Blockstore + Store + Send + Sync,
 {
-    pub fn new(db: DB, chain_config: Arc<ChainConfig>) -> Self
+    pub fn new(
+        db: DB,
+        chain_config: Arc<ChainConfig>,
+        genesis_block_header: BlockHeader,
```

Comment:
Looks like it would be slightly cleaner if this was a reference:
```suggestion
        genesis_block_header: &BlockHeader,
```

Reply:
Done